### DISCUSSION TITLE
how to pick up .tina folder for eslint autofix in webstorm?

### DISCUSSION THREAD

**Discussion by genox** [tinacms/tinacms#3282] [Help]
Hi,

Maybe someone had this issue and could give me a tip on how to solve it.

I can't seem to find a way to add the dotted directory `.tina` to be picked up by my IDEs eslint autofixer and code formatting (Webstorm). There is an option to trigger `eslint --fix` on a glob pattern of files and directories, problem is, there is only one rule that can be set and dotted files and directories seem to be excluded by default.

Is there a way to change Tina's working directory to a non-dotted one? Why is it hidden in the first place? That seems not very intuitive as code changes are necessary and therefore, no need for it to be hidden, technically. no?

I will try to solve the issue with symlinks, maybe that will do the trick. but that's not really nice. I would prefer being able to set the tina directory to be part of my `src` folder instead.

**Comment by jeffsee55** [comment]
We've had a few requests to remove the `.` from the tina folder, it's like that more for historical reasons than anything else. Would love to hear other's input on if they'd prefer just a `tina` folder.

**Reply by coreyaus** [reply]
I think that rename would be an improvement to make Tina more of a first-class citizen in our projects (rather than a hidden folder by default) - as above, it would also make some things slightly more easy when globbing and pattern matching (where hidden files/folders are often ignored by default) 👍

**Reply by genox** [reply]
I just found a pitfall with my previous hack (`ln -s ...`). While it's possible to get an IDE to lint Tina's schema source this way, while building, nextjs picks up the symlink and tries to validate the generated client/admin app. This causes build errors because the nextjs config expects proper typescript sources. Haven't found a way to exclude a directory from building/type checking in nextjs yet.

I use this tsconfig path definition to have nicer imports for types in my source for now :

```
    "paths": {
      "tina/*": ["./.tina/*"],
      ...
    },
```

I wonder how everybody else is handling linting tina schema and imports. Are you just importing from `.tina` via relative paths and linting manually? Or is this some dumb restriction of WebStorm?

**Comment by genox** [comment]
The final solution to this for the moment: To include a dotted folder for eslint, create a file named `.eslintignore` in the root dir and add `!/.tina` to it. ESlint ignores dotted files and dotted directories by default.
